feat: Add a11y tests and GitHub action

ethanWallace opened the pull request:
# Summary | Résumé

Add cypress e2e test to test the accessibility of all pages on the documentation site. Also updated an old GitHub workflow to run these tests on pull request into main and merge into main.

Also includes some accessibility fixes that were identified by the tests, mostly misaligned HTML and headings.

## How to use

### Run tests in browser

- Run `npm run start` to start a local version of the documentation site
- Run `npm run cy:open`
- This will open the cypress interface in the browser of your choice to view the tests

### Run tests in terminal

- Run `npm run cy:run`
- This will start the site and run the tests. The Site will be closed after the tests complete.

https://github.com/cds-snc/design-gc-conception/issues/1192

daine commented:
Looks good, the tests are all successful. However, I can't get the build to finish (that's what's failing on the amplify side).
It's not proceeding to this log on the build process:
```
2025-01-31T18:03:57.709Z [INFO]: # Completed phase: build
## Completed Frontend Build
2025-01-31T18:03:57.718Z [INFO]: ## Build completed successfully
```
the last message is
```
2025-01-31T18:03:57.670Z [INFO]: [11ty] Copied 752 files / Wrote 284 files in 8.94 seconds (31.5ms each, v2.0.1)
```

Locally, here's what I'm getting when I run build:
```
[11ty] Copied 754 files / Wrote 343 files in 754.77 seconds (2200.5ms each, v2.0.1)
2 pages found without an <html> element.
Pages without an outer <html> element will not be processed by default.
If adding this element is not possible, use the root selector config to target a different root element.
```

I wonder what's causing this?

ethanWallace commented:
> [11ty] Copied 754 files / Wrote 343 files in 754.77 seconds (2200.5ms each, v2.0.1)
2 pages found without an <html> element.
Pages without an outer <html> element will not be processed by default.
If adding this element is not possible, use the root selector config to target a different root element.

I was seeing those messages too until I deleted `_site` and ran build again so I don't think that is causing issues on this PR.

ethanWallace commented:
@daine I fixed the deployment issue.

I added a `build:ci` script to the `package.json` to run when someone runs `test` but since it started with `build`, it was also running when someone just ran the build script. That was preventing the deployment from moving onto the next step so that `build:ci` script has been renamed.

melaniebmn commented on src/en/components/radio/base.md:
```diff
@@ -1,6 +1,6 @@
+github: null
 figma: https://www.figma.com/file/mh2maMG2NBtk41k1O1UGHV/GC-Design-System?type=design&node-id=462-110&mode=design&t=juCIOMIg2VKfCrQA-0
 permalink: false
 tags: ['radioEN', 'header']
```
Should we remove the "radio" pages? It's been deprecated for quite a while now.

melaniebmn commented on src/fr/composants/bouton-radio/base.md:
```diff
@@ -1,6 +1,6 @@
+github: null
 figma: https://www.figma.com/file/o4SguSZdar2CCFzSkWNrmB/Syst%C3%A8me-de-design-GC?type=design&node-id=348-5024&mode=design&t=1DaL24vHpjRRfHHm-0
 permalink: false
 tags: ['radioFR', 'header']
```
Same comment - should we remove these pages?

melaniebmn commented on cypress/e2e/a11y-en.cy.js:
```diff
@@ -0,0 +1,75 @@
+    'log',
+    `${violations.length} accessibility violation${
+      violations.length === 1 ? '' : 's'
+    } ${violations.length === 1 ? 'was' : 'were'} detected`,
```
Can we add the `impact` to the log as well, please?

melaniebmn reviewed (approved):
Left a few comments, looks great otherwise! I'm excited to get this merged in :)

melaniebmn reviewed (changes requested):
Left a few comments, looks great otherwise! I'm excited to get this merged in :)

melaniebmn reviewed (approved):
LGTM, excited to start using these tests.

daine reviewed (approved):
Looks great, the output will be very helpful as well! Thank you, LGTM!

<img width="886" alt="Screenshot 2025-02-05 at 10 38 21 PM" src="https://github.com/user-attachments/assets/ba4b8931-6001-4c04-9a3b-c0c784b95859" />
